src/golang/lib/job/process.go

```diff
@@ -146,7 +146,28 @@ func (j *ProcessJobManager) mapJobTypeToCmd(jobName string, spec Spec) (*exec.Cm
 		log.Infof("Logs for job %s are stored in %s", jobName, logFilePath)
 
 		return exec.Command(
-			fmt.Sprintf("%s/%s", j.conf.BinaryDir, workflowExecutorBinary),
+			fmt.Sprintf("%s/%s", j.conf.BinaryDir, exectuorBinary),
+			"--spec",
+			specStr,
+			"--logs-path",
+			logFilePath,
+		), nil
+	} else if spec.Type() == WorkflowRetentionType {
+		workflowRetentionSpec, ok := spec.(*WorkflowSpec)
+		if !ok {
+			return nil, errors.New("Unable to cast job spec to workflowRetentionSpec.")
+		}
+
+		specStr, err := EncodeSpec(workflowRetentionSpec, GobSerializationType)
+		if err != nil {
+			return nil, err
+		}
+
+		logFilePath := path.Join(defaultLogsDir, jobName)
+		log.Infof("Logs for job %s are stored in %s", jobName, logFilePath)
+
+		return exec.Command(
+			fmt.Sprintf("%s/%s", j.conf.BinaryDir, exectuorBinary),
```

Comment:
this is good, can we also move the python connector case out of the `else` at the end here into an `else if`? Lets error if anything falls through to the `else`. This will force us to modify this function appropriately when any new job types are introduced.